frontend/src/pages/RunList.tsx

```diff
@@ -395,9 +398,27 @@ class RunList extends React.PureComponent<RunListProps, RunListState> {
    * the DisplayRun. If the ApiRun has no Pipeline ID, then the corresponding DisplayRun will show
    * '-'.
    */
-  private async _getAndSetPipelineNames(displayRun: DisplayRun): Promise<void> {
+  private async _getAndSetPipelineVersionNames(displayRun: DisplayRun): Promise<void> {
+    const pipelineVersionId = RunUtils.getPipelineVersionId(displayRun.run);
     const pipelineId = RunUtils.getPipelineId(displayRun.run);
-    if (pipelineId) {
+    if (pipelineVersionId) {
+      try {
+        const pipelineVersion = await Apis.pipelineServiceApi.getPipelineVersion(pipelineVersionId);
+        const pipelineVersionName = pipelineVersion.name || '';
+        displayRun.pipelineVersion = {
+          displayName: pipelineVersionName,
+          pipelineId: pipelineVersion && pipelineVersion.resource_references && pipelineVersion.resource_references.find(ref => ref.key && ref.key.type && ref.key.type === ApiResourceType.PIPELINE)!.key!.id!,
```

Comment:
Extract these to a separate util file makes this a lot easier to read.
For example, you can reference: https://github.com/kubeflow/pipelines/blob/5645f25b684d69dc4c997cb010594b094ae24d32/frontend/src/lib/RunUtils.ts#L77

Note that it also handles the case when network data is not valid and returns an `null` gracefully

Reply:
done